File: src/leetcode/2026/05/Solution884.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <sstream>
using namespace std;

class Solution884 {
public:
    vector<string> uncommonFromSentences(string s1, string s2) {
        stringstream ss1(s1), ss2(s2);
        string word;

        unordered_map<string, int> count;
        while (ss1 >> word) {
            count[word]++;
        }
        while (ss2 >> word) {
            count[word]++;
        }
        vector<string> result;
        for (const auto& entry : count) {
            if (entry.second == 1) {
                result.push_back(entry.first);
            }
        }
        return result;
    }
};

int main() {
    Solution884 solution;

    string s1 = "this apple is sweet";
    string s2 = "this apple is sour";
    vector<string> result = solution.uncommonFromSentences(s1, s2);
    for (const string& word : result) {
        cout << word << " ";
    }
    cout << endl; // Output: "sweet sour"

    s1 = "apple apple";
    s2 = "banana";
    result = solution.uncommonFromSentences(s1, s2);
    for (const string& word : result) {
        cout << word << " ";
    }
    cout << endl; // Output: "banana"
    
    return 0;
}

File: src/leetcode/2026/05/Solution884.java
import java.util.*;

public class Solution884 {
    public String[] uncommonFromSentences(String s1, String s2) {
        String[] words1 = s1.split(" ");
        String[] words2 = s2.split(" ");

        Map<String, Integer> countMap = new HashMap<>();

        for (String word : words1) {
            countMap.put(word, countMap.getOrDefault(word, 0) + 1);
        }

        for (String word : words2) {
            countMap.put(word, countMap.getOrDefault(word, 0) + 1);
        }

        List<String> uncommonWords = new ArrayList<>();
        for (Map.Entry<String, Integer> entry : countMap.entrySet()) {
            if (entry.getValue() == 1) {
                uncommonWords.add(entry.getKey());
            }
        }

        return uncommonWords.toArray(new String[0]);
    }

    public static void main(String[] args) {
        Solution884 solution = new Solution884();

        String s1 = "this apple is sweet";
        String s2 = "this apple is sour";
        System.out.println(Arrays.toString(solution.uncommonFromSentences(s1, s2))); // Output: ["sweet", "sour"]

        String s3 = "apple apple";
        String s4 = "banana";
        System.out.println(Arrays.toString(solution.uncommonFromSentences(s3, s4))); // Output: ["banana"]
    }
}
